tests/system_tests_one_router.py

```diff
@@ -3216,5 +3216,86 @@ def run(self):
                 sleep(0.1)
 
 
+class DataConnectionCountTest(TestCase):
+    """
+    Start the router with different numbers of worker threads and make sure
+    that the automatic setting of the number of data connections works
+    correctly. Also make sure that the user can override that automatic
+    setting by providing a specific integer, including zero.
+    """
+    @classmethod
+    def setUpClass(cls):
+        super(DataConnectionCountTest, cls).setUpClass()
+        DataConnectionCountTest.listen_port = cls.tester.get_port()
+        cls.config = Qdrouterd.Config([
+            ('router', {'mode': 'interior', 'id': 'DCC', 'workerThreads': '7'}),
+            ('listener', {'port': DataConnectionCountTest.listen_port, 'role': 'normal'}),
+            ('connector', {'role': 'inter-router', 'host': '127.0.0.1', 'port': DataConnectionCountTest.listen_port, 'saslMechanisms': 'ANONYMOUS', 'idleTimeoutSeconds': '120'}),
+        ])
+
+    def test_60_data_connection_count_absent(self):
+        """
+        If the user omits the dataConnectionCount line from
+        the connector paragraph, it should default to 'auto'.
+        Since this router has worker threads set to '7', we
+        should get (7+1)/2 == 4 data connections.
+        """
+        name = "dcc-test-router"
+        self.router = self.tester.qdrouterd(name, self.config)
```

Comment:
Move these two lines to the `def setUpClass(cls)` function. That way test setup can be done outside of the test

Reply:
I take this comment back. I spoke too soon before looking at the other tests.

Reply:
I did move that name line, though, since they were all the same.
(I hope it's OK that they are all the same?)